File: docs/js/error_handler.js
window.addEventListener("error", function (e) {
	msg  = 'Error Handler\n';
	msg += e.error.message + '\n';
	msg += 'File: ' + e.filename + '\n';
	msg += 'Line: ' + e.lineno + '\n';
	msg += 'Error: ' + e.error + '\n';

	console.error(msg)
	alert(msg);
	return false;
})

function scriptLoadFailure(msg) {
	console.error('Script Load Failure', msg)
	alert('Script Load Failure\n' + msg)
}

window.addEventListener('unhandledrejection', event => {
	console.error('Unhandled Promise Rejection:', event.reason);
	alert("Unandled Promise Rejection: " + event.reason)
	return false;
});

File: docs/error_handler.py
import sys
from js import alert, console

def errorHandler(e):
	exception_type, exception_object, exception_traceback = sys.exc_info()
	filename = exception_traceback.tb_frame.f_code.co_filename
	lineno = exception_traceback.tb_lineno

	msg  = 'Exception Type: ' + str(exception_type) + '\n'
	msg += 'File: ' + filename + '\n'
	msg += 'Line: ' + str(lineno) + '\n'
	msg += str(e)

	console.error(msg)
	alert(msg)

